prometheus/config.go

```diff
@@ -38,6 +38,11 @@ type Configuration struct {
 	// handler on the default HTTP serve mux without listening.
 	ListenAddress string `yaml:"listenAddress"`
 
+	// DynamicListenAddress if specified will be used instead of just registering the
+	// handler on the default HTTP serve mux without listening.
+	// Note: if DynamicListenAddress is specified, ListenAddress is ignored.
+	DynamicListenAddress *ListenAddressConfiguration `yaml:"dynamicListenAddress"`
```

Comment:
Hm this is kinda confusing, e.g., you can have a `dynamicListenAddress` that defines its port in the config file.